src/main/java/com/vmware/vipclient/i18n/base/instances/TranslationMessage.java

```diff
@@ -167,7 +221,7 @@ private String getStringWithArgs(final Locale locale,
                     translation = FormatUtils.formatMsg(translation,
                             LocaleUtility.defaultLocale, (Map<String, Object>) args);
                 } else {
-                    translation = FormatUtils.formatMsg(translation, locale, (Map) args);
+                    translation = FormatUtils.formatMsg(translation, locale, (Map<String, Object>) args);
                 }
             }
```

Comment:
1. How about move these codes to a new private method: getStringWithSource
2. Please try to combine common codes of getStringWithSource and getStringWithoutSource, i think there are some duplicated codes